### ISSUE TITLE
[Linux] Version 7.2.3 requires glibc 2.29+ and is no longer compatible with older distros

### ISSUE THREAD

**Issue by hieucd04** [microsoft/ClearScript#362]
**This issue is only reproducible on version 7.2.3. If I revert to version 7.2.2 everything works OK!**
--

The below exception gets thrown in the logs and causes my website to return 500 for all requests:

> System.TypeLoadException: Cannot load ClearScript V8 library. Load failure information for ClearScriptV8.linux-x64.so:
> 
> /miniskylab/www/nightcore/runtimes/linux-x64/native/ClearScriptV8.linux-x64.so: Unable to load shared library '/miniskylab/www/nightcore/runtimes/linux-x64/native/ClearScriptV8.linux-x64.so' or one of its dependencies. In order to help diagnose loading problems, consider setting the LD_DEBUG environment variable: /miniskylab/www/nightcore/runtimes/linux-x64/native/ClearScriptV8.linux-x64.so: cannot open shared object file: No such file or directory
> 
> /miniskylab/www/nightcore/ClearScriptV8.linux-x64.so: Unable to load shared library '/miniskylab/www/nightcore/ClearScriptV8.linux-x64.so' or one of its dependencies. In order to help diagnose loading problems, consider setting the LD_DEBUG environment variable: /lib64/libm.so.6: version `GLIBC_2.29' not found (required by /miniskylab/www/nightcore/ClearScriptV8.linux-x64.so)
> 
> ClearScriptV8.linux-x64.so: Unable to load shared library 'ClearScriptV8.linux-x64.so' or one of its dependencies.

More info:
- The file `ClearScriptV8.linux-x64.so` does exist at `/miniskylab/www/nightcore/ClearScriptV8.linux-x64.so`
- Environment: CentOS in Docker on Windows via WSL2
- packages I installed:
```
<PackageReference Include="JavaScriptEngineSwitcher.V8" Version="3.17.3" />
<PackageReference Include="Microsoft.ClearScript.V8.ICUData" Version="7.2.3" />
<PackageReference Include="Microsoft.ClearScript.V8.Native.linux-x64" Version="7.2.3" />
<PackageReference Include="Microsoft.ClearScript.V8.Native.win-x64" Version="7.2.3" />
<PackageReference Include="JavaScriptEngineSwitcher.Extensions.MsDependencyInjection" Version="3.3.0" />
```

**Comment by ClearScriptLib** [comment]
Hi @hieucd04,

We build ClearScript's native Linux libraries exactly as before, but it looks like something has changed in V8 9.9 such that it now requires glibc 2.29. Do you know if there's a newer version of CentOS that uses a newer glibc?

Thanks!

**Comment by hieucd04** [comment]
Unfortunately, I'm no Centos expert. So, I don't know!
The version of Centos I'm using is `Centos 8 Stream`
> docker pull quay.io/centos/centos:stream8

**Comment by ClearScriptLib** [comment]
Hi @hieucd04,

We've found a way to build V8 9.9+ and retain support for older versions of glibc, going back at least to Version 2.23. Our next release will restore compatibility with CentOS 8 Stream.

However, it's likely that, at some point in the future, V8 itself will no longer work with such old glibc versions. For our next release, we'll resume testing with Ubuntu 16.04 LTS, which uses glibc 2.23 and is, as far as we know, one of the oldest Linux distributions that .NET supports.

Thanks for reporting this issue!

**Comment by ClearScriptLib** [comment]
Fixed in [Version 7.2.4](https://github.com/microsoft/ClearScript/releases/tag/7.2.4).
